dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java

```diff
@@ -130,73 +130,72 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
 					Date publishDate = (Date) bundle.get("publish_date");
 
 					Logger.info(this, "Processing bundle: ID: " + bundle.get("bundle_id") + ". Status: "
-							+ (UtilMethods.isSet(bundle.get("status")) ? bundle.get("status") : "Starting")
-							+ ". Publish Date: " + publishDate);
-					if ( publishDate.before(new Date()) ) {
-						tempBundleId = (String) bundle.get("bundle_id");
-						MDC.put(BUNDLE_ID, BUNDLE_ID + "=" + tempBundleId);
+						+ (UtilMethods.isSet(bundle.get("status")) ? bundle.get("status") : "Starting")
+						+ ". Publish Date: " + publishDate);
+
+					tempBundleId = (String) bundle.get("bundle_id");
+					MDC.put(BUNDLE_ID, BUNDLE_ID + "=" + tempBundleId);
+
+					try {
+						PushPublishLogger.log(this.getClass(), "Pre-publish work started.");
+						tempBundleContents = pubAPI.getQueueElementsByBundleId(tempBundleId);
+
+						//Setting Audit objects History
+						historyPojo = new PublishAuditHistory();
+						//Retrieving assets
+						Map<String, String> assets = new HashMap<String, String>();
+						List<PublishQueueElement> assetsToPublish = new ArrayList<PublishQueueElement>();
+
+						for ( PublishQueueElement c : tempBundleContents ) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)